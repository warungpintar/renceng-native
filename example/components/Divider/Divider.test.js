import React from 'react';
import {render} from '@testing-library/react-native';
import {Divider} from '@warungpintar/renceng-native';

describe('Divider', () => {
  test('basic, should render without props', () => {
    const {toJSON, queryByTestId} = render(<Divider testID="myDivider" />);
    const style = queryByTestId('myDivider').props.style[0];
    expect(style.borderBottomColor).toBeTruthy();
    expect(style.borderBottomColor).toEqual('black');
    expect(style.borderBottomWidth).toBeTruthy();
    expect(style.borderBottomWidth).toEqual(1);
    expect(style.borderStyle).toBeTruthy();
    expect(style.borderStyle).toEqual('solid');

    expect(toJSON).toMatchSnapshot();
  });

  test('Width', () => {
    const props = {
      size: 4,
      color: 'black',
      type: 'solid',
      testID: 'myDivider',
    };
    const {toJSON, queryByTestId} = render(<Divider {...props} />);
    const style = queryByTestId('myDivider').props.style[0];
    expect(style.borderBottomColor).toBeTruthy();
    expect(style.borderBottomColor).toEqual('black');
    expect(style.borderBottomWidth).toBeTruthy();
    expect(style.borderBottomWidth).toEqual(4);
    expect(style.borderStyle).toBeTruthy();
    expect(style.borderStyle).toEqual('solid');

    expect(toJSON).toMatchSnapshot();
  });

  test('Solid', () => {
    const props = {
      size: 1,
      color: 'black',
      type: 'solid',
      testID: 'myDivider',
    };
    const {toJSON, queryByTestId} = render(<Divider {...props} />);
    const style = queryByTestId('myDivider').props.style[0];
    expect(style.borderBottomColor).toBeTruthy();
    expect(style.borderBottomColor).toEqual('black');
    expect(style.borderBottomWidth).toBeTruthy();
    expect(style.borderBottomWidth).toEqual(1);
    expect(style.borderStyle).toBeTruthy();
    expect(style.borderStyle).toEqual('solid');

    expect(toJSON).toMatchSnapshot();
  });

  test('Dashed', () => {
    const props = {
      size: 1,
      color: 'black',
      type: 'dashed',
      testID: 'myDivider',
    };
    const {toJSON, queryByTestId} = render(<Divider {...props} />);
    const style = queryByTestId('myDivider').props.style[0];
    expect(style.borderBottomColor).toBeTruthy();
    expect(style.borderBottomColor).toEqual('black');
    expect(style.borderBottomWidth).toBeTruthy();
    expect(style.borderBottomWidth).toEqual(1);
    expect(style.borderStyle).toBeTruthy();
    expect(style.borderStyle).toEqual('dashed');

    expect(toJSON).toMatchSnapshot();
  });

  test('Color', () => {
    const props = {
      size: 1,
      color: '#FECF28',
      type: 'dashed',
      testID: 'myDivider',
    };
    const {toJSON, queryByTestId} = render(<Divider {...props} />);
    const style = queryByTestId('myDivider').props.style[0];
    expect(style.borderBottomColor).toBeTruthy();
    expect(style.borderBottomColor).toEqual('#FECF28');
    expect(style.borderBottomWidth).toBeTruthy();
    expect(style.borderBottomWidth).toEqual(1);
    expect(style.borderStyle).toBeTruthy();
    expect(style.borderStyle).toEqual('dashed');

    expect(toJSON).toMatchSnapshot();
  });
});
