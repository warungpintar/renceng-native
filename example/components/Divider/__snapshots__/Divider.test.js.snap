// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Color 1`] = `[Function]`;

exports[`Divider Dashed 1`] = `[Function]`;

exports[`Divider Solid 1`] = `[Function]`;

exports[`Divider Width 1`] = `[Function]`;

exports[`Divider basic, should render without props 1`] = `[Function]`;
